### PR TITLE
dts: 96b_carbon: Fix the model name and compatible

### PR DESCRIPTION
**Pull request by explora26**

### CODE REVIEW

File: dts/arm/96b_carbon.dts [hunk 1]
```diff
@@ -8,8 +8,9 @@
 #include <st/stm32f401.dtsi>
 
 / {
-	model = "STMicroelectronics STM32F401RE-NUCLEO board";
-	compatible = "st,stm32f401re-nucleo", "st,stm32f401";
+	model = "Seeed Studio Carbon 96board";
+	compatible = "seeed,carbon", "st,stm32f401re-nucleo",
```
Comment (mbolivar) [inline dts/arm/96b_carbon.dts]: The `seeed,carbon` part seems to follow the style used in 96b_nitrogen.dts, but shouldn't this be `compatible = "seeed,carbon", "st,stm32f401"` (that is, without the `st,stm32f401re-nucleo`)? 96b_carbon is not pin-compatible with the Nucleo board.

@erwango, is there a policy on how STM-based boards are named here?

Comment (explora26) [inline dts/arm/96b_carbon.dts]: I will remove it, thanks.

File: dts/arm/96b_carbon.dts [hunk 2]
```diff
@@ -8,8 +8,8 @@
 #include <st/stm32f401.dtsi>
 
 / {
-	model = "STMicroelectronics STM32F401RE-NUCLEO board";
-	compatible = "st,stm32f401re-nucleo", "st,stm32f401";
+	model = "Seeed Studio Carbon 96board";
```
Comment (rsalveti) [inline dts/arm/96b_carbon.dts]: Small typo: s/96board/96boards/

**Review by rsalveti — CHANGES_REQUESTED** [review]

File: dts/arm/96b_carbon.dts (hunk 2 above)
Comment (explora26) [inline dts/arm/96b_carbon.dts]: Done, 96b_carbon_nrf51.dts and 96b_nitrogen.dts have typo, too

**Review by rsalveti — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

File: dts/arm/96b_carbon.dts (hunk 1 above)
Comment (erwango) [inline dts/arm/96b_carbon.dts]: @mbolivar,  for now we've been trying to adapt from what was done on Linux. But I agree a real policy is missing.
In the long range, we target to have common dts files Linux/Zephyr/.. , so we need to align with ST guys working on Linux dts files for STM32 and come up with a real policy. Will keep you posted.

**Comment by mbolivar** [comment]
> In the long range, we target to have common dts files Linux/Zephyr/.. , so we need to align with ST guys working on Linux dts files for STM32 and come up with a real policy. Will keep you posted.

Thanks for letting me know.

**Pull request merged**